e2e/portal/tests/ProfilePageActivityLog/ExpandRowsOfActivityOverview.spec.ts

```diff
@@ -19,42 +25,58 @@ import {
 import TableComponent from '@121-e2e/portal/components/TableComponent';
 import LoginPage from '@121-e2e/portal/pages/LoginPage';
 import RegistrationActivityLogPage from '@121-e2e/portal/pages/RegistrationActivityLogPage';
+import RegistrationsPage from '@121-e2e/portal/pages/RegistrationsPage';
 
-let registrationId: number;
-const paymentReferenceId = [registrationPV5.referenceId];
-// Choose the appropriate array based on environment
-const isCI = process.env.CI === 'true';
+const referenceIdPV5 = registrationPV5.referenceId;
+let activitiesCount: number;
 
 // Arrange
 test.beforeEach(async ({ page }) => {
   await resetDB(SeedScript.nlrcMultiple);
 
   const accessToken = await getAccessToken();
   await seedIncludedRegistrations([registrationPV5], programIdPV, accessToken);
-  registrationId = await getRegistrationIdByReferenceId({
-    programId: programIdPV,
-    referenceId: registrationPV5.referenceId,
-    accessToken,
-  });
 
   await doPayment({
     programId: 2,
     paymentNr: 1,
     amount: 100,
-    referenceIds: paymentReferenceId,
+    referenceIds: [referenceIdPV5],
+    accessToken,
+  });
+
+  await waitForPaymentTransactionsToComplete({
+    programId: programIdPV,
+    paymentReferenceIds: [referenceIdPV5],
     accessToken,
+    maxWaitTimeMs: 2_000,
+    completeStatusses: [
+      TransactionStatusEnum.success,
+      TransactionStatusEnum.waiting,
+    ],
   });
 
   await updateRegistration(
     2,
-    registrationPV5.referenceId,
+    referenceIdPV5,
     {
       maxPayments: '2',
     },
     'automated test',
     accessToken,
   );
 
+  await waitForMessagesToComplete({
+    programId: programIdPV,
+    referenceIds: [referenceIdPV5],
+    accessToken,
+    minimumNumberOfMessagesPerReferenceId: 3,
+  });
+
+  activitiesCount = (
+    await getAllActivitiesCount(programIdPV, referenceIdPV5, accessToken)
+  ).totalCount;
```

Comment:
nice that this works, but I'm not fully getting it. Can you explain me? It somehow feels like a weird mix-up of back-end testing that shouldn't have to be part of such e2e tests..